### PR TITLE
Feature/api calls

### PR DESCRIPTION
**Pull request by knees4bees**
## Is this a fix, feature, or refactor?
Feature

## What is the change?
* Adds linting and resolves several linting warnings/errors.
* Adds error handling for network requests.
* Creates and styles new `Error` component.

## What does it fix?
n/a

## Where should the reviewer start?
Look at `App.js` and `utilities.js` and note the new flow (`utilities` now just holds the fetch calls themselves; two new state properties are in `App.js`; there is a new `renderError` method that calls a new `Error` component). There is definitely room for refactoring here -- which is something we'll want to do on the PR we tag the instructors on (see the project spec). 

## How should this be tested?
* Linting: run `npm run lint`. If the linter causes the app to crash (this may happen when you run `npm start`), you may need to navigate to `node_modules` > `react-scripts` > `config` > `webpack.config.js`, look for the `ESLintPlugin` section near line 750, and set `failOnError: false`. (There is probably a better way to do this??)
* Error handling: first confirm that the happy path is still happy. Then uncomment line 57 in `App.js` and change the status code in the parentheses to whatever you'd like. An appropriate error message should appear on the screen.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,49 +1,109 @@
...
+        this.setState({ fetchStatus: response.status });
+
+        if (!response.ok) {
+          throw new Error();
+        }
+
+        return response.json();
+      })
+      .then((movieData) => this.setState({ selectedMovieDetails: movieData.movie }))
+      .catch((err) => this.setState( {fetchError: true }));
```
Comment (itsnameissteven) [inline src/App.js:49]: Smart having a fetchStatus as state. I think this whole thing is really cleaver and changes the way I imaged error handling, great idea. Could this be refactored into a reusable function?

File: src/App.js [hunk 2]
```diff
@@ -1,49 +1,109 @@
...
   resetHome = () => {
     this.setState({ selectedMovieDetails: {} });
   }
 
   render() {
+    // return this.renderError(404); // To force render of error state
+
+    if (this.state.fetchError) {
+      return this.renderError(this.state.fetchStatus);
+    }
```
Comment (itsnameissteven) [inline src/App.js:61]: is renderError a react method?

File: src/App.js [hunk 3]
```diff
@@ -1,49 +1,109 @@
...
+        break;
+      case 404:
+        message = 'Page not found.'
+        break;
+      case 422:
+        message = 'Invalid request.'
+        break;
+      default:
+        message = 'Oops! Something went wrong. Please try again.';
+    }
```
Comment (itsnameissteven) [inline src/App.js:98]: ohhh nice. this is the first switch I've seen in the wild!

File: src/components/Carousel/Carousel.js [hunk 4]
```diff
@@ -32,13 +31,12 @@ const Carousel = ({movies, displayMovieDetail}) => {
   };
 
   return (
-    <div className='carousel'>
-      <Slider {...settings}>  
+    <div className="carousel">
```
Comment (itsnameissteven) [inline src/components/Carousel/Carousel.js:34]: I'm assuming the linter wants double quotes for html elements from now on???

File: src/components/MovieDetail/MovieDetail.js [hunk 5]
```diff
@@ -1,40 +1,42 @@
...
   const ratingStyle = {
     background: `linear-gradient(to right, #ffd000 ${percentage}%, #cfe2f7d8 ${percentage}%)`
-  }
-  
+  };
+
   return (
-    <div className='movie-detail'>
-      <section className='movie-card__details' >
+    // if (details) {vv that stuff}
```
Comment (itsnameissteven) [inline src/components/MovieDetail/MovieDetail.js:12]: What is VV?

**Review by itsnameissteven — COMMENTED** [review]
This looks great, Thanks for tackling all the error handling, The linter will be great for keeping us consistent

File: src/App.js (hunk 2 above)
Comment (knees4bees) [reply src/App.js:61]: It is now! :)

File: src/components/MovieDetail/MovieDetail.js (hunk 5 above)
Comment (knees4bees) [reply src/components/MovieDetail/MovieDetail.js:12]: LOL sorry this was a pseudocode line I left in by accident.

File: src/App.js (hunk 1 above)
Comment (knees4bees) [reply src/App.js:49]: Yes, let's try to refactor it!

File: src/components/Carousel/Carousel.js (hunk 4 above)
Comment (knees4bees) [reply src/components/Carousel/Carousel.js:34]: Yep

**Pull request merged**